src/rgw/rgw_op.cc

```diff
@@ -579,6 +583,73 @@ int rgw_build_object_policies(RGWRados *store, struct req_state *s,
   return ret;
 }
 
+void rgw_add_to_iam_environment(rgw::IAM::Environment& e, std::string&& key, std::string&& val){
+  e.emplace(std::piecewise_construct,
+	    std::forward_as_tuple(key),
+	    std::forward_as_tuple(val));
```

Comment:
`key` and `val` are lvalues in this statement, so they need to be cast with `std::move()` to enable move construction

and the `piecewise_construct` overload isn't necessary when both constructors take a single argument (see https://github.com/ceph/ceph/pull/18706)

Reply:
dropping this, there doesn't seem places (yet) where we can call std::move on both the key and value in the current patchset.